javascript/ql/src/semmle/javascript/frameworks/jQuery.qll

```diff
@@ -258,144 +258,152 @@ private class JQueryChainedElement extends DOM::Element, InvokeExpr {
+    override DataFlow::Node getADataNode() { result = getOptionArgument([0 .. 1], "data") }
 
-  private string getResponseType() {
-    getOptionArgument([0 .. 1], "dataType").mayHaveStringValue(result)
-  }
+    private string getResponseType() {
+      getOptionArgument([0 .. 1], "dataType").mayHaveStringValue(result)
+    }
 
-  override DataFlow::Node getAResponseDataNode(string responseType, boolean promise) {
-    (
-      responseType = getResponseType()
-      or
-      not exists(getResponseType()) and responseType = ""
-    ) and
-    promise = false and
-    (
-      result =
-        getOptionArgument([0 .. 1], "success")
-            .getALocalSource()
-            .(DataFlow::FunctionNode)
-            .getParameter(0)
-      or
-      result =
-        getAResponseNodeFromAnXHRObject(getOptionArgument([0 .. 1],
-            any(string method | method = "error" or method = "complete"))
+    override DataFlow::Node getAResponseDataNode(string responseType, boolean promise) {
+      (
+        responseType = getResponseType()
+        or
+        not exists(getResponseType()) and responseType = ""
+      ) and
+      promise = false and
+      (
+        result =
+          getOptionArgument([0 .. 1], "success")
               .getALocalSource()
               .(DataFlow::FunctionNode)
-              .getParameter(0))
-      or
-      result = getAnAjaxCallbackDataNode(this)
-      or
-      result = getAResponseNodeFromAnXHRObject(getAnXHRObject(this))
-    )
+              .getParameter(0)
+        or
+        result =
+          getAResponseNodeFromAnXHRObject(getOptionArgument([0 .. 1],
+              any(string method | method = "error" or method = "complete"))
+                .getALocalSource()
+                .(DataFlow::FunctionNode)
+                .getParameter(0))
+        or
+        result = getAnAjaxCallbackDataNode(this)
+        or
+        result = getAResponseNodeFromAnXHRObject(getAnXHRObject(this))
+      )
+    }
   }
-}
-
-/**
- * Gets the response data node from a call to a jqXHR Object.
- */
-DataFlow::Node getAnAjaxCallbackDataNode(ClientRequest::Range request) {
-  result =
-    request.getAMemberCall(any(string s | s = "done" or s = "then")).getCallback(0).getParameter(0)
-}
-
-/**
- * Gets the `jqXHR` object from a call to `fail` on the result from an ajax call (`request`).
- */
-DataFlow::SourceNode getAnXHRObject(ClientRequest::Range request) {
-  result = request.getAMemberCall("fail").getCallback(0).getParameter(0)
-}
 
-/**
- * Gets a node refering to the response contained in an `jqXHR` object (`obj`).
- */
-DataFlow::SourceNode getAResponseNodeFromAnXHRObject(DataFlow::SourceNode obj) {
-  result =
-    obj
-        .getAPropertyRead(any(string s |
-            s = "responseText" or
-            s = "responseXML"
-          ))
-}
+  /**
+   * Gets a response data node from a call to a method on jqXHR Object.
```

Comment:
> That is a reference to the request parameter of the predicate.
> I'll change it.

I see. How about mentioning the parameter like this then? (We very rarely put the predicate parameter names in parentheses, so that is what tripped me)
```suggestion
   * Gets a response data node from a call to a method on jqXHR Object `request`.
```